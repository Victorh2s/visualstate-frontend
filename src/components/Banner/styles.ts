import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`

  width: 100%;
  height: 22rem;
  overflow: hidden;
  cursor: pointer;


  `}
`;

export const Banner = styled.img`
  ${({ theme }) => css`
  width: 100%;

  `}
`;

export const GladientVertical = styled.div`
  ${({ theme }) => css`


  width: inherit;
  height: inherit;
  background: linear-gradient(to bottom, #0d0c0c 5%, transparent 100%);


  `}
`;

export const DivButtons = styled.div`
  ${({ theme }) => css`

  margin-top: -3rem;
  float: right;
  width: 10%;
  height:3rem;
  border-top-left-radius: 10px;
  background: ${theme.colors.dark};
  text-align: center;
  cursor: pointer;

  > svg {
    color: ${theme.colors.white};
  `}
`;

export const Edit = styled.button`
  ${({ theme }) => css`
  border: none;
  width: 3rem;
  height:3rem;
  background: transparent;
  cursor: pointer;

  > svg {
    color: ${theme.colors.white};
  `}
`;
