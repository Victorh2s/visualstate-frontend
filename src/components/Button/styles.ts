import styled, { css } from 'styled-components';
import { ButtonProps } from '.';

export const Button = styled.button<Pick<ButtonProps, 'color'>>`
  ${({ theme, color }) => css`
    background: ${theme.colors[color]};
    border: none;
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.small};
    padding: ${theme.spacings.small} ${theme.spacings.medium};
    cursor: pointer;
    border-radius: 1rem;
    transition: all 300ms ease-in-out;
    display: flex;
    align-items: center;
    justify-content: center;

    &:focus {
      outline: none;
      box-shadow: 0 0 ${theme.spacings.small} ${theme.colors[color]};
      filter: brightness(110%);
    }

    &:hover {
      filter: brightness(90%);
    }

    &:disabled {
      background: ${theme.colors.gray8};
      color: ${theme.colors.white};
      cursor: not-allowed;

      &:hover {
        filter: none;
      }
    }

    > svg {
      width: 2rem;
      height: 2rem;
      margin-left: 1rem;
    }
  `}
`;
