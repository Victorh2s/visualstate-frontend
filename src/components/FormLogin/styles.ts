import styled, { css } from 'styled-components';

export const Form = styled.form`
`;

export const ButtonWrapper = styled.div``;

export const Container = styled.div`
    text-align: center;
    width: 35vw;
`;

export const Wrapper = styled.div`
  ${({ theme }) => css`
    width: 100%;
    margin-bottom: ${theme.spacings.large};

  `}
`;

export const InputWrapper = styled.div`
  ${({ theme }) => css`
    position: relative;

    > svg {
      position: absolute;
      top: 50%;
      transform: translate(0, -50%);
      right: 1rem;
      width: 2.5rem;
      height: 2.5rem;
      color: ${theme.colors.gray6};
      z-index: ${theme.layers.layer1};

    }

    *:focus ~ svg {
      color: ${theme.colors.primary};

    }

  `}
`;

export const Input = styled.input`
  ${({ theme }) => css`
    border: 1px solid ${theme.colors.gray3};
    width: 100%;
    height: 100%;
    font-size: ${theme.font.sizes.normal};
    transition: ${theme.transitions.fast};
    padding: ${theme.spacings.small} ${theme.spacings.xsmall};
    background: ${theme.colors.white};
    padding-right: 3.7rem;
    border-radius: ${theme.spacings.tiny};
    outline: none;

    &::placeholder {
      visibility: hidden;
      opacity: 0;
    }

    &:focus {
      border: ${theme.spacings.xxtiny} solid ${theme.colors.primary};
      box-shadow: 0 0 ${theme.spacings.xtiny} 0 ${theme.colors.primary};
      background: ${theme.colors.white};
    }

    &:focus
      + ${Label},
      &:not(:placeholder-shown)
      + ${Label},
      &:-webkit-autofill
      + ${Label} {
      top: 0;
      transform: translate(0, -50%);
      font-size: ${theme.font.sizes.xsmall};
      color: ${theme.colors.white};
      background: ${theme.colors.primary};
      filter: none;
    }


  `}
`;

export const Label = styled.label<{ element: string }>`
  ${({ theme, element }) => css`
    display: inline;
    position: absolute;
    top: 50%;
    transform: translate(0, -50%);
    left: ${theme.spacings.xsmall};
    font-size: ${theme.font.sizes.small};
    height: ${theme.font.sizes.small};
    transition: ${theme.transitions.fast};
    background: ${theme.colors.white};
    padding: 0.5rem 1rem;
    display: flex;
    align-items: center;
    line-height: 1;
    z-index: ${theme.layers.layer1};
    color: ${theme.colors.gray6};
    border-radius: ${theme.spacings.tiny};

    ${
      element === 'textarea' &&
      css`
      top: ${theme.spacings.large};
      transform: translate(0, -50%);
    `
    }
  `}
`;

export const DivButton = styled.div`

`;
export const DivLink = styled.div`
  ${({ theme }) => css`

    margin-bottom: ${theme.spacings.large};
    > a{
      text-decoration: none;
    }
  `}
`;

export const button = styled.button`
  ${({ theme }) => css`

    background: ${theme.colors.primary};
    border: none;
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.small};
    padding: ${theme.spacings.small} ${theme.spacings.medium};
    cursor: pointer;
    border-radius: 1rem;
    transition: all 300ms ease-in-out;
    display: flex;
    margin: 0 auto;

    &:focus {
      outline: none;
      box-shadow: 0 0 ${theme.spacings.small} ${theme.colors.primary};
      filter: brightness(110%);
    }

    &:hover {
      filter: brightness(90%);
    }



    > svg {
      width: 2rem;
      height: 2rem;
      margin-left: 1rem;
    }
  `}
`;

export const ErrorMessage = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.warning};
    margin: ${theme.spacings.xxsmall} 0 ${theme.spacings.medium};
  `}
`;
