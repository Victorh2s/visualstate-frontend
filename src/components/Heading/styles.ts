import styled, { css, DefaultTheme } from 'styled-components';
import { HeadingProps } from '.';

const titleSize = {
  supersmall: (theme: DefaultTheme) => css`
  font-size: ${theme.font.sizes.xxsmall};
    ${mediaFont(theme)};
  `,
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.medium};
    ${mediaFont(theme)};
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.large};
    ${mediaFont(theme)};
  `,
  big: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xlarge};
    ${mediaFont(theme)};
  `,
  huge: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xhuge};
    ${mediaFont(theme)};
  `,
};

const mediaFont = (theme: DefaultTheme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.medium};
  }
  @media ${theme.media.lteSmall} {
    font-size: ${theme.font.sizes.small};
  }
  @media ${theme.media.lteSuperSmall} {
    font-size: ${theme.font.sizes.xxsmall};
  }
`;

const titleCase = (uppercase: boolean) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1<HeadingProps>`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.dark : theme.colors.white};
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
  `}
`;
