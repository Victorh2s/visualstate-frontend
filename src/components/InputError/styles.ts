import styled, { css } from 'styled-components';

export const ErrorMessage = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.warning};
    margin: ${theme.spacings.xxsmall} 0 ${theme.spacings.large};
  `}
`;
