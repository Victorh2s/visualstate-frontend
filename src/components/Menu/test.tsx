import { fireEvent, screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { Menu } from '.';
import { theme } from '../../styles/theme';

describe('<Menu />', () => {
  it('should render the button', () => {
    renderTheme(<Menu menuid={false} />);

    expect(screen.getByLabelText('Open/Close menu')).toBeInTheDocument();
  });

  it('should render text', () => {
    renderTheme(<Menu menuid={false} />);

    expect(screen.getByText('VisualState')).toBeInTheDocument();
    expect(screen.getByText('Entrar')).toBeInTheDocument();
    expect(screen.getByText('Criar Conta')).toBeInTheDocument();
  });

  it('Testing functions of button', () => {
    renderTheme(<Menu menuid={false} />);

    const button = screen.getByLabelText('Open/Close menu');
    const menuContainer = button.nextSibling;

    expect(button).toHaveStyleRule('display', 'none');
    expect(button).toHaveStyleRule('display', 'flex', {
      media: theme.media.lteOrEqMedium,
    });

    fireEvent.click(button);
    expect(menuContainer).toHaveStyleRule('left', '-32rem', {
      media: theme.media.lteOrEqMedium,
    });
    expect(screen.getByLabelText('Close menu')).toBeInTheDocument();

    fireEvent.click(menuContainer);
    expect(menuContainer).toHaveStyleRule('left', '-32rem', {
      media: theme.media.lteOrEqMedium,
    });
    expect(screen.getByLabelText('Open menu')).toBeInTheDocument();
  });
});

it('should match snapshot', () => {
  const { container } = renderTheme(<Menu menuid={false} />);

  expect(container).toMatchSnapshot();
});
