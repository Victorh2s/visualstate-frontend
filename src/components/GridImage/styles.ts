import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
${({ theme }) => css`
padding: 2rem;
padding-bottom: 10rem;
 > ${TextComponent}{
  text-align: center;
  margin-bottom: ${theme.spacings.xhuge}
}
${HeadingContainer}{
  text-align: center;
}
`}
`;

export const Grid = styled.div`
${({ theme }) => css`
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(280px, 1fr));
  gap: ${theme.spacings.large}
`}
`;

export const GridElement = styled.div`
  overflow: hidden;
`;

export const Image = styled.img`
  width: 100%;
  transition: all 300ms ease-in-out;
  &:hover{
    transform: scale(1.2);
  }
`;
