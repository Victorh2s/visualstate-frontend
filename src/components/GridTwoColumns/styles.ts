import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export type translatey = {
  translatey: number;
};

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    overflow-x: hidden;
    align-items: center;
    width: 100%;
    gap: ${theme.spacings.large};
    @media ${theme.media.lteOrEqMedium} {
      display: block;
      text-align: center;
      align-items: center;
      overflow-x: hidden;
    }
    @media ${theme.media.lteOrEqSmallest} {
      width: 70%;
  }
    ${Title} {
      margin-bottom: 0;
    }
  `}
`;

export const TextContainer = styled.div`
    width: 35rem;
    display: flex;
    overflow-x: hidden;
    margin: 0 auto;
    padding-bottom: 0.1rem;
`;

export const Ytext = styled.div<translatey>`
  ${({ theme, translatey }) => css`
    display: flex;
    transform: translateX(-${translatey}rem);
    transition: all 400ms ease-in-out;
    padding-bottom: 0.1rem;
    @media ${theme.media.lteOrEqSmallest} {
      transform: translateX(-${translatey - 11}rem);
      transition: all 400ms ease-in-out;
    }
`}
`;

export const HeadingHeight = styled.div`
 ${({ theme }) => css`
  width: 35rem;
  padding-bottom: 0.1rem;
  @media ${theme.media.lteOrEqSmallest} {
      width: 25rem;
  }
  `}
`;

export const Carrousel = styled.div`
 ${({ theme }) => css`
  height: 40rem;
  margin: 0 auto;
  overflow-y: hidden;
  @media ${theme.media.lteOrEqSmallest} {
    height: 25rem;
  }

  `}
`;

export const Yvideo = styled.div<translatey>`
  ${({ theme, translatey }) => css`
    transform: translateY(-${translatey}rem);
    transition: all 400ms ease-in-out;
    @media ${theme.media.lteOrEqMedium} {
      transform: translateY(-${translatey + 12}rem);
      transition: all 400ms ease-in-out;
  }

  `}
`;

export const SourceContainer = styled.div`
  width: 100%;
  height: 50rem;
`;

export const videos = styled.video`
  ${({ theme }) => css`
  width: 45vw;
  height: 50rem;
  @media ${theme.media.lteOrEqMedium} {
    width: 80%;
    height: 50rem;
    align-items: center;
  }
  @media ${theme.media.lteOrEqSmallest} {
    width: 60%;
    height: 50rem;
    align-items: center;
  }
    `}
`;
