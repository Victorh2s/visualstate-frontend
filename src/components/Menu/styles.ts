import styled, { css, DefaultTheme } from 'styled-components';
import { Container as SectionContainer } from '../SectionContainer/styles';
import { Title as Heading } from '../Heading/styles';

type Background = {
  Background: boolean;
  Visible: boolean;
};
type MenuBehaviorProps = {
  Visible: boolean;
};

const menuVisible = (
  Visible: MenuBehaviorProps['Visible'],
  theme: DefaultTheme,
) => css`
  overflow-y: ${Visible ? 'auto' : 'hidden'};
  left: ${Visible ? '0' : '-30rem'};
  @media ${theme.media.lteMedium} {
    left: ${Visible ? '0' : '-32rem'};
  }
`;

export const Container = styled.div<Background>`
  ${({ theme, Visible, Background }) => css`
    position: fixed;
    z-index: 5;
    top: 0;
    right: 0;
    left: 0;
    width: 100%;
    border-bottom: ${theme.colors.mediumGray};
    transition: all 300ms ease-in-out;
    background: ${Background ? '#07090D' : 'transparent'};

    > ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }
    & ${Heading} {
      margin-top: 0;
      margin-bottom: 0;
    }
    @media ${theme.media.lteMedium} {
      ${Visible && menuVisible(Visible, theme)};
      display: block;
      position: fixed;
      text-align: center;
      background: ${theme.colors.dark};
      padding: ${theme.spacings.xxlarge} 0;
      max-width: 32rem;
      height: 100rem;
      overflow-y: auto;
      right: 0;
      left: -32rem;
      transition: all 300ms ease-in-out;
      > ${SectionContainer} {
        padding-top: 0;
        padding-bottom: 0;
      }
      & ${Heading} {
      margin-top: 0;
      margin-bottom: 5rem;
      }

    }
  `}
`;

export const MenuContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;
    align-items: center;
    @media ${theme.media.lteMedium} {
      display: block;
      text-align: center;
      padding: ${theme.spacings.xxlarge} 0;
    }
  `}
`;

const buttonChanger = (
  Visible: MenuBehaviorProps['Visible'],
  theme: DefaultTheme,
) => css`
   right: ${Visible ? '26rem' : '1rem'};
   color: ${Visible ? theme.colors.white : theme.colors.white};
   @media ${theme.media.lteMedium} {
    left: ${Visible ? '26rem' : '1rem'};
  }
 `;

export const Button = styled.button<MenuBehaviorProps>`
  ${({ theme, Visible }) => css`
    z-index: 6;
    position: fixed;
    top: 2rem;
    right: 2rem;
    width: 4rem;
    height: 4rem;
    background: transparent;
    color: ${theme.colors.white};
    border: none;
    display: none;
    cursor: pointer;
    pointer-events: ${Visible ? 'none' : 'all'};
    @media ${theme.media.lteMedium} {
      display: flex;
      align-items: center;
      justify-content: center;

    }
    ${buttonChanger(Visible, theme)}
    > svg {
      width: 2.5rem;
      height: 2.5rem;
    }
  `}
`;
