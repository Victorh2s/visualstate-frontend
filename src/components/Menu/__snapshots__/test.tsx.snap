// Vitest Snapshot v1

exports[`should match snapshot 1`] = `
.c4 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 0 auto;
}

.c7 {
  color: #07090D;
  font-size: 2.4rem;
  text-transform: none;
}

.c2 {
  position: fixed;
  z-index: 5;
  top: 0;
  right: 0;
  left: 0;
  width: 100%;
  border-bottom: #DDDDDD;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  background: transparent;
}

.c2 > .c3 {
  padding-top: 0;
  padding-bottom: 0;
}

.c2 .c6 {
  margin-top: 0;
  margin-bottom: 0;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  z-index: 6;
  position: fixed;
  top: 2rem;
  right: 2rem;
  width: 4rem;
  height: 4rem;
  background: transparent;
  color: #FFFFFF;
  border: none;
  display: none;
  cursor: pointer;
  pointer-events: all;
  right: 1rem;
  color: #FFFFFF;
}

.c0 > svg {
  width: 2.5rem;
  height: 2.5rem;
}

.c8 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #FFFFFF;
}

.c8 > img {
  height: 3rem;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.c10 >a {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #FFFFFF;
  position: relative;
}

.c10 >a::after {
  content: '';
  position: absolute;
  left: 50%;
  bottom: 0.75rem;
  width: 0;
  height: 0.2rem;
  background: #FFFFFF;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c10 >a:hover::after {
  left: 25%;
  width: 50%;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

@media (max-width:768px) {
  .c7 {
    font-size: 2.4rem;
  }
}

@media (max-width:414px) {
  .c7 {
    font-size: 1.6rem;
  }
}

@media (max-width:362px) {
  .c7 {
    font-size: 1rem;
  }
}

@media (max-width:768px) {
  .c2 {
    display: block;
    position: fixed;
    text-align: center;
    background: #07090D;
    padding: 4.0rem 0;
    max-width: 32rem;
    height: 100rem;
    overflow-y: auto;
    right: 0;
    left: -32rem;
    -webkit-transition: all 300ms ease-in-out;
    transition: all 300ms ease-in-out;
  }

  .c2 > .c3 {
    padding-top: 0;
    padding-bottom: 0;
  }

  .c2 .c6 {
    margin-top: 0;
    margin-bottom: 5rem;
  }
}

@media (max-width:768px) {
  .c5 {
    display: block;
    text-align: center;
    padding: 4.0rem 0;
  }
}

@media (max-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:768px) {
  .c0 {
    left: 1rem;
  }
}

@media (max-width:768px) {
  .c9 {
    -webkit-flex-flow: column wrap;
    -ms-flex-flow: column wrap;
    flex-flow: column wrap;
    -webkit-align-content: center;
    -ms-flex-line-pack: center;
    align-content: center;
  }
}

<div>
  <button
    aria-label="Open/Close menu"
    class="c0"
  >
    <svg
      aria-hidden="true"
      aria-label="Open menu"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <div
    aria-hidden="false"
    class="c2"
  >
    <div
      class="c3 c4"
    >
      <div
        class="c5"
      >
        <h1
          class="c6 c7"
        >
          <a
            class="c8"
            href="#"
          >
            VisualState
          </a>
        </h1>
        <nav
          aria-label="Main menu"
          class="c9"
        >
          <div
            class="c10"
          >
            <a
              href="/"
            >
              Entrar
            </a>
          </div>
          <div
            class="c10"
          >
            <a
              href="/cadastro"
            >
              Criar Conta
            </a>
          </div>
        </nav>
      </div>
    </div>
  </div>
</div>
`;
